Engine/Fuego/Renderer/Buffer.cpp

```diff
@@ -0,0 +1,52 @@
+#include "Buffer.h"
+
+#include "Renderer.h"
+
+namespace Fuego
+{
+VertexBuffer* VertexBuffer::Create(float* vertices, uint32_t size)
+{
+    UNUSED(vertices);
+    UNUSED(size);
+
+    switch (Renderer::GetAPI())
```

Comment:
I think it is incorrect approach. We need to employ polymorphism instead of switch-case.